robot-name/tests/robot-name.rs

```diff
@@ -11,8 +11,9 @@ impl Robot {
 */
 
 fn assert_name_matches_pattern(n: &str) {
-    assert!(n[0..2].chars().all(|c| c >= 'A' && c <= 'Z'), "name starts with 2 letters");
-    assert!(n[3..].chars().all(|c| c >= '0' && c <= '9'), "name ends with 3 numbers");
+    assert!(n.len() == 5, "name is exactly 5 characters long");
+    assert!(n[0..2].chars().all(|c| c >= 'A' && c <= 'Z'), "name starts with 2 uppercase letters");
+    assert!(n[2..].chars().all(|c| c >= '0' && c <= '9'), "name ends with 3 numbers");
```

Comment:
The previous test was ignoring the 3rd char. This was from the previous
commit that was supposed to change the format from AAA123 to AA123, but
didn't move indices for both tests.

Secondly, the (3..) slice happily returned and used an empty result if the
string was too short, passing the digits test.

On Tuesday, September 8, 2015, Eduardo Bautista <notifications@github.com>
wrote:

> In robot-name/tests/robot-name.rs
> <https://github.com/exercism/xrust/pull/47#discussion_r38983094>:
>
> > @@ -11,8 +11,9 @@ impl Robot {
> >  */
> >
> >  fn assert_name_matches_pattern(n: &str) {
> > -    assert!(n[0..2].chars().all(|c| c >= 'A' && c <= 'Z'), "name starts with 2 letters");
> > -    assert!(n[3..].chars().all(|c| c >= '0' && c <= '9'), "name ends with 3 numbers");
> > +    assert!(n.len() == 5, "name is exactly 5 characters long");
> > +    assert!(n[0..2].chars().all(|c| c >= 'A' && c <= 'Z'), "name starts with 2 uppercase letters");
> > +    assert!(n[2..].chars().all(|c| c >= '0' && c <= '9'), "name ends with 3 numbers");
>
> So the previous test was expecting the name to end with 4 numbers?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/exercism/xrust/pull/47/files#r38983094>.
>